pkg/cli/helm/helm.go

```diff
@@ -101,7 +101,24 @@ func helmChartFromContainerRegistry(version string, config *helm.Configuration,
 
 	pull.DestDir = dir
 
-	_, err = pull.Run(releaseName)
+	var chartRef string
+
+	if !registry.IsOCI(repoUrl) {
+		// For non-OCI registries (like contour), we need to set the repo URL
+		// to the registry URL. The chartRef is the release name.
+		// ex.
+		// pull.RepoURL = https://charts.bitnami.com/bitnami
+		// pull.Run("contour")
+		pull.RepoURL = repoUrl
+		chartRef = releaseName
+	} else {
+		// For OCI registries (like radius), we will use the
+		// repo URL + the releaseName as the chartRef.
+		// pull.Run("oci://ghcr.io/radius-project/helm-chart/radius")
+		chartRef = fmt.Sprintf("%s/%s", repoUrl, releaseName)
+	}
+
+	_, err = pull.Run(chartRef)
```

Comment:
So in the Run function, if the RepoURL is empty (else condition above) then the logic will just go and get the details from the ChartRef?

Reply:
exactly. Setting the RepoURL to an OCI repo did not work when I tried it. setting it here as the chartRef did though.